python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -812,10 +812,12 @@ def merge(*def_sets: "Definitions") -> "Definitions":
 
                 component_tree = def_set.component_tree
 
-            check.invariant(
-                not def_set.has_resolved_repository_def(),
-                "Definitions object should have been resolved",
-            )
+            # This check is commented out now, since it's possible that particular component
+            # Definitions are resolved by another component
+            # check.invariant(
+            #     not def_set.has_resolved_repository_def(),
+            #     "Definitions object should have been resolved",
+            # )
```

Comment:
tell me more about whats going on here? im skeptical that we want to leave in commented code like this

Reply:
`Definitions.merge` expects that no child definitions have been resolved already. In e.g. `DefsFolderComponent`, we merge the `Definitions` objects for all child components. Some of these may have been previously resolved when e.g. they were referenced by another component cross-stack, causing this invariant to fail.

Reply:
got it, I think putting something closer to your response here as committed comment is better than the commented out invariant call

Reply:
i personally am not sure what the right move is

Reply:
moved out to https://github.com/dagster-io/dagster/pull/31178